src/Analysis/GWASV2/Utils/StateManagement/reducer.js

```diff
@@ -45,6 +45,15 @@ const reducer = (state, action) => {
     return { ...state, selectedHare: action.payload };
   case ACTIONS.UPDATE_FINAL_POPULATION_SIZES:
     return { ...state, finalPopulationSizes: action.payload };
+  case ACTIONS.ADD_MESSAGE:
+    return { ...state, messages: [...state.messages, action.payload] };
+  case ACTIONS.DELETE_MESSAGE:
+    return {
+      ...state,
+      messages: state.messages.filter(
```

Comment:
I noticed this approach, like the card approach we have, leads to there being duplicate messages being shown (for example I got this to render pretty quick) but closing one of them closes them all. Do we want to add a conditional to the add message action so duplicate error messages don't display (i.e. only add the message if it is not in state.messages )

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/113449836/213499911-498ef8b6-3937-411c-a3bd-b951088e6419.png">

Reply:
> add a conditional to the add message action so duplicate error messages don't display

I think that is feasible 👍

Reply:
done.